packages/ui-tree-browser/src/TreeBrowser/props.ts

```diff
@@ -31,28 +31,87 @@ import type {
+   */
   defaultExpanded?: (string | number)[]
+  /**
+   * There are 2 types of tree selection:  single and multi.
+   * This is set up to allow for "multi" in the future without having to deprecate the old API.
+   */
   selectionType?: 'none' | 'single'
-  size?: 'small' | 'medium' | 'large'
-  variant?: 'folderTree' | 'indent'
-  collectionIcon?: React.ReactNode | ((...args: any[]) => any)
-  collectionIconExpanded?: React.ReactNode | ((...args: any[]) => any)
-  itemIcon?: React.ReactNode | ((...args: any[]) => any)
-  getItemProps?: (...args: any[]) => any
-  getCollectionProps?: (...args: any[]) => any
+  onCollectionToggle?: (collection: CollectionData) => void
+  onItemClick?: (data: CollectionData) => void
+  /**
+   * Whether or not to show the root collection specified in rootId prop or
+   * to begin with its immediate subcollections and items instead
+   */
   showRootCollection?: boolean
-  onCollectionClick?: (...args: any[]) => any
-  onCollectionToggle?: (...args: any[]) => any
-  onItemClick?: (...args: any[]) => any
+  /**
+   * An optional label to assist visually impaired users
+   */
   treeLabel?: string
-  renderContent?: (...args: any[]) => any
+  /**
+   * An optional compare function to specify order of the collections and the items
+   */
   sortOrder?: (obj1: any, obj2: any) => number
+} & TreeBrowserBaseProps
+
+// props shared between TreeBrowser, TreeCollection
+type TreeBrowserBaseProps = {
+  /**
+   * A function called with each item's props as an argument. The return value of this function is a
+   * props object which will be passed to the item when it is rendered. This is useful for situations where
+   * you need to render the item differently depending on it's props. For example, if you would like to
+   * display a different icon for items with a certain name.
+   */
+  getItemProps?: (props: Record<string, any>) => Record<string, any> // cant use generics here :/
+  /**
+   * A function called with each collection's props as an argument. The return value of this function is a
+   * props object which will be passed to the collection when it is rendered. This is useful for situations where
+   * you need to render the collection differently depending on it's props. For example, if you would like to
+   * display a different icon for collections with a certain name.
+   */
+  getCollectionProps?: (props: Record<string, any>) => Record<string, any>
+  onCollectionClick?: (e: React.MouseEvent, data: CollectionData) => void
+} & TreeBrowserCommonProps
+
+// props shared between TreeBrowser, TreeCollection, TreeButton
+type TreeBrowserCommonProps = {
+  size?: 'small' | 'medium' | 'large'
+  variant?: 'folderTree' | 'indent'
+  collectionIcon?: React.ReactNode | ((props: unknown) => React.ReactNode)
+  collectionIconExpanded?:
+    | React.ReactNode
+    | ((props: unknown) => React.ReactNode)
+  itemIcon?: React.ReactNode | ((props: unknown) => React.ReactNode)
+  renderContent?: (props: any) => JSX.Element
```

Comment:
Isn't the props here `TreeBrowserButtonProps`? It seems to be used in TreeButton's render method.

Reply:
yep, fixed